app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use App\Models\Department;
+use App\Models\User;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\RegisterUserRequest;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Hash;
+
+class RegisterController extends Controller
+{
+    public function index()
+    {
+        return view('auth.register', ['departments' => Department::all()]);
+    }
+
+    /**
+     * Register a new user
+     *
+     * @param RegisterUserRequest $request
+     * @return Factory|View|\Illuminate\Http\RedirectResponse
+     */
+    public function register(RegisterUserRequest $request)
+    {
+        $validatedUserData = $request->validated();
+
+        $user = User::create([
+            'username' => $validatedUserData->username,
```

Comment:
see the usage of `$request->validated()` method.
it simply returns list of all data defined in RegisterUserRequest class.

Rather than mapping the validated one by one, it simply returns the set of all validated data